service/server.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"context"
+	"io"
+	"net"
+	"sync"
+
+	common "github.com/xlabs/tss-common"
+	"github.com/xlabs/tss-common/service/signer"
+	"github.com/xlabs/tss-lib/v2/tss"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+type Server struct {
+	signer.UnimplementedSignerServer
+	ctx    context.Context
+	cancel context.CancelFunc
+	logger *zap.Logger
+
+	*grpc.Server
+	tss.Signer
+	listener net.Listener
+
+	waitersLock  sync.Mutex
+	waiters      map[uint64]chan<- *signer.SignResponse
+	nextWaiterID uint64
+}
+
+// addSubscriber creates a channel for a new subscriber, adds it to the waiters map,
+// and returns the subscriber's ID and channel.
+func (s *Server) addSubscriber() (uint64, chan *signer.SignResponse) {
+	ch := make(chan *signer.SignResponse)
```

Comment:
Should probably buffer this channel to avoid losing responses in fanoutSignatures.